modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/liferay.js

```diff
@@ -14,6 +14,64 @@
 
 Liferay = window.Liferay || {};
 
+Liferay.Webpack = {
+	Container: {},
+
+	SharedScope: {},
+
+	require: function(moduleName, callback) {
+		var parts = moduleName.split("/");
+
+		var containerId = parts[0];
+		var path =
+			parts.length == 1 ?
+				'.' :
+				"./" + parts.slice(1).join("/");
+
+		var container = Liferay.Webpack.Container[containerId];
+
+		var resolveRequire = function() {
+			console.log('Looking up module', moduleName, '...');
+			Promise.resolve(container.get(path)).then(module => {
+
+				console.log('Resolved module:', moduleName);
+				callback(module());
+			})
+		};
+
+		if (container) {
+			resolveRequire()
+		} else {
+			var script = document.createElement("script");
+
+			script.src = "/o/" + containerId + "/webpack/container.js";
+
+			script.onload = function() {
+				container = Liferay.Webpack.Container[containerId];
+
+				if(!container) {
+					console.error(
+						new Error(
+							'Container ' + containerId + ' was fetched but ' +
+								'did not define itself in ' +
+								'Liferay.Webpack.Container'));
+					return;
+				}
+
+				console.log('Initializing container', containerId, '...');
+				Promise.resolve(
+					container.init(Liferay.Webpack.SharedScope)
+				).then(_ => {
+					resolveRequire()
+				});
```

Comment:
So `resolveRequire()` might throw, right? (Because `callback` might throw), which means we might wind up with an unhandled promise rejection. Would be could to have a `catch()` for that; what do you think?

Reply:
I think we will move this to a Promise based API in the end, as @jbalsas and me discussed in another place (which I cannot find to link it :-( ).